ruby_on_rails/advanced_forms_and_activerecord/installing_psql.md

```diff
@@ -0,0 +1,239 @@
+
+<div class="lesson-note" markdown="1">
+#### Note
+We will be using `sudo` quite a bit. As you follow these instructions, you may be (re-)prompted for your password. Remember that there will be no visual feedback when you type in your password.
+</div>
+
+To update our system, type in these commands:
+
+~~~~bash
+sudo apt update
+sudo apt upgrade
+~~~~
+
+Be sure to review any prompts before pressing `Y` and `Enter`. While they typically advise you about additional space required for updates, this might not be the case!
+
+### Step 2: Install the PostgreSQL packages
+After our system is up to date, we will install the packages for PostgreSQL.
+
+~~~bash
+sudo apt install postgresql postgresql-contrib libpq-dev
+~~~
+
+After installation is complete, let's start the server using this command:
+
+~~~bash
+sudo systemctl start postgresql.service
+~~~
+
+If you are unsure about whether postgre is active, it's possible to check with this command:
+
+~~~bash
+service postgresql status
+~~~
+
+Got an error, or don't see an active service? Come visit the Discord for some help!
+
+If postgre is active, you can press `Q` to quit the status screen and move on to the next step.
+
+### Step 3: Setting up PostgreSQL
+PostgreSQL is now running, but we have to configure it in order to be able to use it with our local Rails applications.
+
+#### 3.1 PostgreSQL Roles
+PostgreSQL authenticates via roles. A role is like a user and the default installation has set up a `postgres` role that we can use to interact with the database server. This is great, but that would mean having to switch to that role every time we wanted to do something with the database server.
+
+Instead, we will set up our own user to avoid switching to `postgres` all the time.
+
+<div class="lesson-note" markdown="1">
+#### IMPORTANT
+You'll want to take note of your Linux username so we can create the right role! The terminal prompt, by default, is configured in this format:
+
+~~~bash
+<linux_username>@<machine_name>: 
+~~~
+</div>
+
+#### 3.2 Creating a New Role
+With all your information ready, let's create a role in Postgre! The command to do so is:
+
+~~~bash
+sudo -u postgres createuser --interactive
+~~~
+
+Make sure that you enter your Linux user name as the name of the role, and be sure to make that user a superuser. Setting up a role this way means that we leverage "peer authentication" which makes using the local database very easy from Rails.
+
+#### 3.3 Creating the Role Database
+One other important step in setting up Postgre is that each role must have its own database of the same name. Without it, the role we just created will not be able to log in or interact with the Postgre server.
+
+You can try to run `psql` now, but you will get an error that the database does not exist. Not to worry, let's create one to resolve fix this:
+
+~~~bash
+sudo -u postgres createdb <linux_username>
+~~~
+
+Now our role is fully set up: we've got `<role_name>` and that role has a database.
```

Comment:
Just noting that I ended up with such terminal outputs but the thing worked perfectly still:
![console weirdness](https://user-images.githubusercontent.com/95084627/195979910-35bce47c-32b5-4c05-87c4-edf84564e658.png)

Reply:
Thanks for this, I'm checking if this is a critical error now.

I am going to fiddle around with `sudo -i -u postgres <command>` as it has to do with using the `postgres` user, and the two linux users can't access each other's home directories.

Will report back.

Reply:
Adding the `-i` option to the `sudo` commands removes the directory error, so I will update the instructions to use that flag as well.